client/src/components/filterCampaines/index.jsx

```diff
@@ -0,0 +1,58 @@
+import axios from 'axios';
+import PropTypes from 'prop-types';
+import React, { useEffect, useState } from 'react';
+import {
+  message,
+  Radio,
+  Input,
+  Segmented,
+
+} from 'antd';
+import './style.less';
+
+const { Search } = Input;
+function Compaigns({ setCategory, setAvailable, setSearch }) {
+  const [categories, setCategories] = useState([]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const { data: { data: { categories: categoriesFromDB } } } = await axios('/api/categories');
+        const categoriesNameArr = ['List All'];
+        categoriesFromDB.map((item) => categoriesNameArr.push(item.name));
+        setCategories(categoriesNameArr);
+      } catch ({ response: { data: { message: errorMessage } } }) {
+        message.info(errorMessage);
+      }
+    };
+    fetchData();
```

Comment:
What this line do?

Reply:
it fetches all the categories so we display them in front